src/functionBased/components/TodoItem.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect } from 'react';
+import { FaTrash } from 'react-icons/fa';
+import PropTypes from 'prop-types';
+import styles from './TodoItem.module.css';
+
+const TodoItem = (props) => {
+  const [editing, setEditing] = useState(false);
+
+  const handleEditing = () => {
+    setEditing(true);
+  };
+
+  const handleUpdatedDone = (event) => {
+    if (event.key === 'Enter') {
+      setEditing(false);
+    }
+  };
+
+  const completedStyle = {
+    fontStyle: 'italic',
+    color: '#595959',
+    opacity: 0.4,
+    textDecoration: 'line-through',
+  };
+
+  const {
+    todo, handleChangeProps, deleteTodoProps, setUpdate,
+  } = props;
+
+  const viewMode = {};
+  const editMode = {};
+
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  useEffect(
+    () => () => {
+      console.log('Cleaning up...');
+    },
```

Comment:
I also suggest you avoid the use of console.log() in your code per the JavaScript best practices.

Reply:
Thanks a lot for your suggestions @amon-cofie . I will do well to implement them.